### ISSUE TITLE
Multiple libraries that share a download url

### ISSUE THREAD

**Issue by edkimmel** [mapbox/mason#391]
This is regarding packages like the android ndk and boost, where multiple libraries end up downloading the same file.

Is there any way to have the cache files be based on url, or to override the slug? My goal is to have packages like this all share the same cache file instead of redownloading it under a different name.

Boost isn't as bad, since the download is only ~85mb, but downloading the entire ndk for each architecture hurts.

**Comment by springmeyer** [comment]
Yes, this has bothered me about boost as well. The download and untar is done for every specific library build and not shared. Fixing this is not urgent but would be nice to make development faster.

> Is there any way to have the cache files be based on url, or to override the slug? 

There is not a way currently. We'd need to rework the logic inside `mason.sh`.
